src/Administration/Resources/app/administration/src/module/sw-settings-custom-field/component/sw-custom-field-detail/index.js

```diff
@@ -165,8 +165,8 @@ Component.register('sw-custom-field-detail', {
             }
 
             this.currentCustomField.config = {
-                ...this.currentCustomField.config,
                 ...this.fieldTypes[customFieldType].config,
+                ...this.currentCustomField.config,
```

Comment:
Can you add or adjust the existing test cases here: `Resources/app/administration/test/module/sw-settings-custom-field/component/sw-custom-field-detail.spec.js`

i'm wondering why it wasn't catched by the existing tests? Probably it is some edge case to the problem occurs only for some field types?

Reply:
I can customize the test, it just takes a while since I haven't worked with Jest too much yet.

> i'm wondering why it wasn't catched by the existing tests? Probably it is some edge case to the problem occurs only for some field types?
The current test only checks the type select.
Furthermore, there are (currently!) only additional default configs (numberType) for the number type, which can also be changed afterwards by the corresponding editor component.

But it doesn't change the fact, that the order of the merge is wrong.
You always have to use the default values first, which will be overwritten with the current values.

Reply:
I have started writing the test(s).

I first split the current test. The current test first started with type select, then switched to switch and then tested saving with it. The relevant method applyTypeConfiguration was also never executed there or tested separately.
So the test was neither complete nor correct before.

I started with appropriate adjustments, but currently come to my limits with Jest, as I do not have too much experience with it. Any help or guidance would be appreciated.

My problem:
After changing the customFieldType to number, the type specific editor component sw-custom-field-type-number is loaded with a select for the numberType (int|float).
Accessing the component is quite difficult, the find method only finds the select with a simple (and rather inaccurate) selector, but was eventually able to find it.
But when I change the value, this change is not applied.

Reply:
Not entirely sure what's the issue there, but maybe that the change is applied async, maybe you can try `await Vue.nextTick();` as described here: https://v1.test-utils.vuejs.org/guides/testing-async-components.html